src/java/com/rapleaf/jack/queries/GenericQueryBuilder.java

```diff
@@ -0,0 +1,184 @@
+import com.rapleaf.jack.Column;
+import com.rapleaf.jack.Table;
+import com.rapleaf.jack.queries.where_operators.IWhereOperator;
+
+public class GenericQueryBuilder {
+  private static final Logger LOG = LoggerFactory.getLogger(GenericQueryBuilder.class);
+
+  private static int MAX_CONNECTION_RETRIES = 1;
+
+  private final GenericQuery genericQuery;
+  private final BaseDatabaseConnection dbConnection;
+
+  public GenericQueryBuilder(BaseDatabaseConnection dbConnection, GenericQuery genericQuery) {
+    this.dbConnection = dbConnection;
+    this.genericQuery = genericQuery;
+  }
+
+  public JoinConditionBuilder leftJoin(Table table) {
+    return new JoinConditionBuilder(this, JoinType.LEFT_JOIN, table);
+  }
+
+  public JoinConditionBuilder rightJoin(Table table) {
+    return new JoinConditionBuilder(this, JoinType.RIGHT_JOIN, table);
+  }
+
+  public JoinConditionBuilder innerJoin(Table table) {
+    return new JoinConditionBuilder(this, JoinType.INNER_JOIN, table);
+  }
+
+  GenericQueryBuilder addJoinCondition(JoinCondition joinCondition) {
+    genericQuery.addJoinCondition(joinCondition);
+    return this;
+  }
+
+  public GenericQueryBuilder where(Column column, IWhereOperator operator) {
+    genericQuery.addWhereCondition(new WhereConstraint(column, operator, null));
+    return this;
+  }
+
+  public GenericQueryBuilder andWhere(Column column, IWhereOperator operator) {
+    genericQuery.addWhereCondition(new WhereConstraint(column, operator, WhereConstraint.Logic.AND));
+    return this;
+  }
+
+  public GenericQueryBuilder orWhere(Column column, IWhereOperator operator) {
+    genericQuery.addWhereCondition(new WhereConstraint(column, operator, WhereConstraint.Logic.OR));
+    return this;
+  }
+
+  public GenericQueryBuilder orderBy(Column column, QueryOrder queryOrder) {
+    genericQuery.addOrderCondition(new OrderCriterion(column, queryOrder));
+    return this;
+  }
+
+  public GenericQueryBuilder orderBy(Column column) {
+    genericQuery.addOrderCondition(new OrderCriterion(column, QueryOrder.ASC));
+    return this;
+  }
+
+  public GenericQueryBuilder limit(int offset, int limit) {
+    genericQuery.addLimitCondition(new LimitCriterion(offset, limit));
+    return this;
+  }
+
+  public GenericQueryBuilder limit(int limit) {
+    genericQuery.addLimitCondition(new LimitCriterion(limit));
+    return this;
+  }
+
+  public GenericQueryBuilder groupBy(Column column, Column... columns) {
+    genericQuery.addGroupByColumns(column, columns);
+    return this;
+  }
+
+  public GenericQueryBuilder select(Column column, Column... columns) {
+    genericQuery.addSelectedColumns(column, columns);
+    return this;
+  }
+
+  public String getFullSqlStatement() throws IOException {
+    return getPreparedStatement().toString();
+  }
+
+  public Records fetch() throws IOException {
+    int retryCount = 0;
+    PreparedStatement preparedStatement = getPreparedStatement();
+    LOG.info("query: {}", preparedStatement.toString());
```

Comment:
Yeah, it'll be way too verbose. Just make it debug level and it won't be logged by default.